packages/evasive-transform/src/transform-comment.js

```diff
@@ -46,3 +46,44 @@ export function transformComment(node, unmapLoc) {
     unmapLoc(node.loc);
   }
 }
+
+/**
+ * Inspects a comment for a hint that it must be preserved by a transform.
+ *
+ * @param {string} comment
+ */
+const markedForPreservation = comment => {
+  if (comment.startsWith('!')) {
+    return true;
+  }
+  if (comment.startsWith('*')) {
+    // Detect jsdoc style @preserve, @copyright, @license, @cc_on (IE
+    // cconditional comments)
+    return /(?:^|\n)\s*\*?\s*@(?:preserve|copyright|license|cc_on)\b/.test(
```

Comment:
This pattern is unnecessarily vulnerable to ReDoS.
```suggestion
    return /(?:^|\n)\s*(?:\*\s*)?@(?:preserve|copyright|license|cc_on)\b/.test(
```

Reply:
Thanks, I didn't consider regexp efficiency.

Some other ideas:
https://claude.site/artifacts/badb117f-f454-4e8c-bcf9-8037160acb97
https://chatgpt.com/share/e/da63d95f-f9e6-4f28-86b5-96d80be0fe37

Reply:
For increased availability, the claude suggestion is:

```js
const optimizedRegex = /^[\s*]*@(?:preserve|copyright|license|cc_on)\b/;
```

I can't seem to access the chatgpt suggestion.

Reply:
Claude Haiku is free to use at a rate limit. Fwiw, I think it's reasonable that `@copyright` anywhere in the comment should be considered an intent to preserve.

Reply:
Added the relaxed suggestion above, but FWIW @kriskowal's intuition is pretty much correct. The most problematic input for the merged regular expression is lots of whitespace followed by an asterisk and then not an at sign, and I think the backtracking is exponential:
```console
$ esbench.mjs -b1 -h 'V8' --arg i~3 \
  -s '
    const S = " ".repeat(10**(i + 1));
    const sMatch = `${S}*@preserve`;
    const sNoMatch = `${S}*X`, r = /(?:^|\n)\s*\*?\s*@(?:preserve|copyright|license|cc_on)\b/;
  ' \
  accept:'if(!r.test(sMatch)) throw Error("unexpected rejection");' \
  reject:'if(r.test(sNoMatch)) throw Error("unexpected acceptance")'
#### V8
accept (0) 7491.209439528024 ops/ms after 31 245760-count observations
reject (0) 3480 ops/ms after 29 122880-count observations
accept (1) 4924.580152671756 ops/ms after 21 245760-count observations
reject (1) 89.9080919080919 ops/ms after 34 2647-count observations
accept (2) 1298.868 ops/ms after 34 38202-count observations
reject (2) 1.2203389830508475 ops/ms after 34 36-count observations
accept (3) 154.3412228796844 ops/ms after 34 4603-count observations
reject (3) 0.0125 ops/ms after 13 1-count observations
```